src/bloom.cpp

```diff
@@ -242,9 +264,20 @@ bool CRollingBloomFilter::contains(const std::vector<unsigned char>& vKey) const
     return b1.contains(vKey);
 }
 
-void CRollingBloomFilter::clear()
+bool CRollingBloomFilter::contains(const uint256& hash) const
 {
-    b1.clear();
-    b2.clear();
+    if (nInsertions < nBloomSize / 2) {
+        return b2.contains(hash);
+    }
+    return b1.contains(hash);
+}
+
+void CRollingBloomFilter::reset(unsigned int nNewTweak)
+{
+    if (!nNewTweak)
+        nNewTweak = GetRand(std::numeric_limits<unsigned int>::max());
```

Comment:
By not bother, I mean that I am reasonably confident that the current code will not cause any problems, as the global initialization is done single-threaded.

Despite that, I think it's cleaner to not consume randomness in the constructor.